pkg/controller/nodenetworkstate/nodenetworkstate_controller.go

```diff
@@ -113,10 +115,20 @@ func (r *ReconcileNodeNetworkState) Reconcile(request reconcile.Request) (reconc
 			return err
 		}
 
-		err = nmstate.UpdateCurrentState(r.client, instance)
+		stateToReport, err := nmstate.CurrentState()
+		if err != nil {
+			setConditionDegraded(instance, err.Error())
```

Comment:
Don't we want to do similar stuff like we do for policies, moving this into s`statestatus` package and also do the retry on conflict there for status update (although I think there is no conflicts here).

Reply:
There should be no conflict. If we get a bug with a conflict, it is a red flag.

I want to keep this minimal, unlit the other conditions, here there is no concurency, we don't update while reconciling, we just use this helper function to set fields in the Status instance. Save it later together with updated spec. I'm happy with this whole controller being a single short module, which it is now.